documenteer/bin/refreshlsstbib.py

```diff
@@ -0,0 +1,119 @@
+from future.standard_library import install_aliases
+install_aliases()  # NOQA
+
+import argparse
+import io
+import logging
+import os
+import sys
+
+import requests
+
+from .._version import get_versions
+__version__ = get_versions()['version']
+del get_versions
+
+
+def run():
+    """Command line entrypoint for the ``refresh-lsst-bib`` program.
+    """
+    args = parse_args()
+
+    if args.verbose:
+        log_level = logging.DEBUG
+    else:
+        log_level = logging.INFO
+    logging.basicConfig(
+        level=log_level,
+        format='%(asctime)s %(levelname)s %(name)s: %(message)s')
+
+    logger = logging.getLogger(__name__)
+
+    logger.info('refresh-lsst-bib version {0}'.format(__version__))
+
+    error_count = process_bib_files(args.dir)
+
+    sys.exit(error_count)
+
+
+def parse_args():
+    """Create an argument parser for the ``refresh-lsst-bib`` program.
+
+    Returns
+    -------
+    args : `argparse.Namespace`
+        Parsed argument object.
+    """
+    parser = argparse.ArgumentParser(
+        description="Download LSST .bib bibliography files from the "
+                    "lsst-texmf GitHub repository.",
+        epilog="Version {0}".format(__version__)
+    )
+    parser.add_argument(
+        '-d', '--dir',
+        default='.',
+        help="Directory to download bib files into. Default is the current "
+             "workig directory.")
+    parser.add_argument(
+        '-v', '--verbose',
+        dest='verbose',
+        action='store_true', default=False,
+        help='Enable Verbose output (debug level logging)'
+    )
+    return parser.parse_args()
+
+
+def process_bib_files(local_dir):
+    """Run the refresh-lsst-bib program's logic: iterates through bib URLs,
+    downloads the file from GitHub, and writes it to a local directory.
+
+    Parameters
+    ----------
+    local_dir : `str`
+        Directory to write bib files into.
+
+    Returns
+    -------
+    error_count : int
+        Number of download errors.
+    """
+    logger = logging.getLogger(__name__)
+
+    # check the output directory exists
+    if not os.path.isdir(local_dir):
+        logger.error('Output directory "{}" does not exist'.format(local_dir))
+        sys.exit(1)
+
+    root_blob_url = 'https://raw.githubusercontent.com/lsst/lsst-texmf/master/texmf/bibtex/bib/'  # noqa
+    bib_filenames = ['books.bib', 'lsst-dm.bib', 'lsst.bib', 'refs.bib',
+                     'refs_ads.bib']
+
+    error_count = 0
+    for bib_filename in bib_filenames:
+        url = root_blob_url + bib_filename
+        logger.info('Downloading {}'.format(url))
+        try:
+            content = get_content(url)
+        except requests.HTTPError as e:
+            logger.exception(str(e))
+            logger.warning('Could not download {}'.format(url))
+            error_count += 1
+            continue
+
+        local_filename = os.path.join(local_dir, bib_filename)
+        with io.open(local_filename, 'w') as f:
```

Comment:
why use `io.open` and not:
```python
with open(filename, 'w') as f:
   f.write(content)
```
Please add a comment here to make it more obvious as to the reason.